core/src/services/yandex_disk/backend.rs

```diff
@@ -0,0 +1,328 @@
+        self.core.ensure_dir_exists(to).await?;
+
+        let resp = self.core.move_object(from, to).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::OK | StatusCode::CREATED => {
+                resp.into_body().consume().await?;
+
+                Ok(RpRename::default())
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn copy(&self, from: &str, to: &str, _args: OpCopy) -> Result<RpCopy> {
+        self.core.ensure_dir_exists(to).await?;
+
+        let resp = self.core.copy(from, to).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::OK | StatusCode::CREATED => {
+                resp.into_body().consume().await?;
+
+                Ok(RpCopy::default())
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn read(&self, path: &str, _args: OpRead) -> Result<(RpRead, Self::Reader)> {
+        let download_url = self.core.get_download_url(path).await?;
+
+        let req = Request::get(download_url)
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+        let resp = self.core.send(req).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::OK => {
+                let size = parse_content_length(resp.headers())?;
+                let range = parse_content_range(resp.headers())?;
+                Ok((
+                    RpRead::new().with_size(size).with_range(range),
+                    resp.into_body(),
+                ))
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn stat(&self, path: &str, _args: OpStat) -> Result<RpStat> {
+        let resp = self.core.metainformation(path, None, None).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::OK => {
+                let bs = resp.into_body().bytes().await?;
+
+                let mf: MetainformationResponse =
+                    serde_json::from_slice(&bs).map_err(new_json_deserialize_error)?;
+
+                parse_info(mf).map(RpStat::new)
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn write(&self, path: &str, _args: OpWrite) -> Result<(RpWrite, Self::Writer)> {
+        let writer = YandexDiskWriter::new(self.core.clone(), path.to_string());
+
+        let w = oio::OneShotWriter::new(writer);
+
+        Ok((RpWrite::default(), w))
+    }
+
+    async fn delete(&self, path: &str, _: OpDelete) -> Result<RpDelete> {
+        let resp = self.core.delete(path).await?;
+
+        let status = resp.status();
+
+        match status {
+            StatusCode::OK => Ok(RpDelete::default()),
+            StatusCode::NO_CONTENT => Ok(RpDelete::default()),
+            StatusCode::ACCEPTED => Ok(RpDelete::default()),
```

Comment:
When will yandex returns accepted for delete?

Reply:
When delete a large folder

Reply:
> When delete a large folder

Is this an asynchronous operation? If so, it could impact our process: once the delete command is executed, the file or folder should no longer exist.

Reply:
It is an asynchronous operation

Reply:
GOt it, let's revisit this part in the future. Would you like to add some comment about this behavior in code?

Reply:
fixed